### PR TITLE
[helm create] Add service account to templates

### LINKED ISSUE #5879: helm create - Make it easier to do the right thing, part i: serviceaccount
Hello,

Helm create should add options to create `serviceaccount`, enabled by default imho, as it is a known best practice to run every workload with its own service account, by including the option to create a service account, we are making it easier to do the right thing.

cc @bacongobbler @unguiculus @scottrigby @rimusz

### PR DESCRIPTION
**Pull request by naseemkullah**
closes #5879 

**If applicable**:
- [x] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

### CODE REVIEW

File: pkg/chartutil/create.go [hunk 1]
```diff
@@ -238,6 +248,15 @@ spec:
     app.kubernetes.io/name: {{ include "<CHARTNAME>.name" . }}
     app.kubernetes.io/instance: {{ .Release.Name }}
 `
+const defaultServiceAccount = `{{- if .Values.serviceAccount.create -}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "<CHARTNAME>.serviceAccountName" . }}
+  labels:
+  {{ include "<CHARTNAME>.labels" . | indent 4 }}
```
Comment (bacongobbler) [inline pkg/chartutil/create.go]: this should probably use `nindent` instead.

Comment (mattfarina) [reply pkg/chartutil/create.go]: `nindent` adds a new line at the start. But `{{` is used at the start of this to the prior whitespace is kept including a new line. I think `indent` works but to be consistent with the other files created the two spaces before the `{{` need to be removed.

Comment (naseemkullah) [reply pkg/chartutil/create.go]: Thanks!

**Review by mattfarina — APPROVED** [review]
For reference, this pattern is common in the charts repo.

**Comment by naseemkullah** [comment]
Thanks @mattfarina , whats the Needs V3 fix label about? Does this feature have to be duplicated elsewhere?

**Comment by bacongobbler** [comment]
this PR isn't relevant for Helm 3 so I'm marking this as "complete".

**Comment by bacongobbler** [comment]
ported to Helm 3 in https://github.com/helm/helm/pull/6490

**Pull request merged**